src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -64,6 +64,10 @@
  * {@code strictfp} modifier when using JDK 17 or later. See reason at
  * <a href="https://openjdk.org/jeps/306">JEP 306</a>
  * </li>
+ * <li>
+ * {@code final} modifier on unnamed variables and unnamed pattern variables
+ * when using JDK 21 or later.
```

Comment:
Question: Is this 21 preview or actually 21?

Reply:
21 was a first preview

Reply:
So then like another PR, shouldn't this say 22 or specifically mention 21 previous and released in 22?

Reply:
@nrmancuso @romani Does this create an issue as the logic in the check also actually says 21. We have no preview property.

Reply:
@rnveach good point, I think it makes sense to use the Java version when the feature was officially released

Reply:
done to 22. accordingly default changed to 22 also